client/app/src/main/res/layout/activity_user_profile.xml

```diff
@@ -1,5 +1,55 @@
 <?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="match_parent" android:layout_height="match_parent">
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
 
-</RelativeLayout>
\ No newline at end of file
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:orientation="vertical"
+        app:layout_constraintBottom_toTopOf="@id/bottom_navigation_frag"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <include
+            android:id="@+id/component_profile"
+            layout="@layout/component_profile"
+            android:layout_height="wrap_content"
+            android:layout_width="match_parent" />
+
+        <TextView
+            android:id="@+id/user_books_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginEnd="16dp"
+            android:ems="10"
+            android:text="@string/user_books_text"
+            android:textSize="24sp"
+            app:layout_constraintTop_toBottomOf="@id/component_profile" />
+
+        <ListView
+            android:id="@+id/user_books_listview"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:divider="@null"
+            android:dividerHeight="0dp"
+            android:focusable="true"
+            app:layout_constraintTop_toBottomOf="@id/user_books_text" />
+    </LinearLayout>
+
+    <fragment

```

Comment:
Also can we prevent the bottom nav fragment from reacting to the keyboard? @AdityaHarvi  had this issue as well

Reply:
I think you can fix this by putting the fragment inside the linear layout and having the ListView fill up the remaining space